airflow/contrib/operators/gcp_transfer_operator.py

```diff
@@ -0,0 +1,753 @@
+
+    def _reformat_date(self, field_key):
+        schedule = self.body['schedule']
+        if field_key not in schedule:
+            return
+        if isinstance(schedule[field_key], date):
+            schedule[field_key] = self._convert_date_to_dict(schedule[field_key])
+
+    def _reformat_time(self, field_key):
+        schedule = self.body['schedule']
+        if field_key not in schedule:
+            return
+        if isinstance(schedule[field_key], time):
+            schedule[field_key] = self._convert_time_to_dict(schedule[field_key])
+
+    def _reformat_schedule(self):
+        if 'schedule' not in self.body:
+            return
+        self._reformat_date('scheduleStartDate')
+        self._reformat_date('scheduleEndDate')
+        self._reformat_time('startTimeOfDay')
+
+    def process_body(self):
+        self._inject_aws_credentials()
+        self._reformat_schedule()
+        return self.body
+
+    @staticmethod
+    def _convert_date_to_dict(field_date):
+        """
+        Convert native python ``datetime.date`` object  to a format supported by the API
+        """
+        return {'day': field_date.day, 'month': field_date.month, 'year': field_date.year}
+
+    @staticmethod
+    def _convert_time_to_dict(time):
+        """
+        Convert native python ``datetime.time`` object  to a format supported by the API
+        """
+        return {"hours": time.hour, "minutes": time.minute, "seconds": time.second}
+
+
+class TransferJobValidator:
+    def __init__(self, body):
+        self.body = body
+
+    def _verify_data_source(self):
+        is_gcs = 'gcsDataSource' in self.body['transferSpec']
+        is_aws_s3 = 'awsS3DataSource' in self.body['transferSpec']
+        is_http = 'httpDataSource' in self.body['transferSpec']
+
+        sources_count = sum([is_gcs, is_aws_s3, is_http])
+        if sources_count != 0 and sources_count != 1:
+            raise AirflowException(
+                "More than one data source detected. Please choose exactly one data source from: "
+                "gcsDataSource, awsS3DataSource and httpDataSource."
+            )
+
+    def _restrict_aws_credentials(self):
+        if 'awsS3DataSource' not in self.body['transferSpec']:
+            return
+
+        if 'awsAccessKey' in self.body['transferSpec']['awsS3DataSource']:
+            raise AirflowException(
+                "AWS credentials detected inside the body parameter (awsAccessKey). This is not allowed, "
+                "please use Airflow connections to store credentials."
+            )
+
+    def _restrict_empty_body(self):
+        if not self.body:
+            raise AirflowException("The required parameter 'body' is empty or None")
+
+    def validate_body(self):
+        self._restrict_empty_body()
+
+        if 'transferSpec' not in self.body:
+            return
+
+        self._restrict_aws_credentials()
+        self._verify_data_source()
+
+
+class GcpTransferServiceJobCreateOperator(BaseOperator):
```

Comment:
The latest recommendations say that we should keep one file for one service.  Airflow is growing very fast at the moment and this is the current way to organize operators. The community is working on other ways to organize operators, but it will be a long process. For now, we can not split operators to multiple files.

I will now add comments that separate the various operator groups.

Reference: 
https://docs.google.com/document/d/1_rTdJSLCt0eyrAylmmgYc3yZr-_h51fVlnvMmWqhCkY/edit?ts=5bb72dfd#